runtime/src/waitable_condvar.rs

```diff
@@ -44,29 +44,44 @@ pub mod tests {
         let cv_ = cv.clone();
         let cv2_ = cv2.clone();
         let cv2__ = cv2.clone();
+        let waiters_active = Arc::new(AtomicUsize::new(0));
+        let waiters_active_ = waiters_active.clone();
+        let waiters_active__ = waiters_active.clone();
         // several passes to check re-notification and drop one of the
         let passes = 3;
         let handle = Builder::new().spawn(move || {
             for _pass in 0..passes {
                 let mut notified = false;
+                let mut timed_out = false;
+                waiters_active_.fetch_add(1, Ordering::SeqCst);
                 while cv2_.wait_timeout(Duration::from_millis(1)) {
```

Comment:
Can these condition variables wake up spuriously and return false for the `timed_out()` check? https://doc.rust-lang.org/nightly/std/sync/struct.Condvar.html#method.wait_timeout